src/test/scala/strawman/collection/test/TraverseTest.scala

```diff
@@ -69,9 +69,8 @@ class TraverseTest {
 
   @Test
   def eitherSequenceTest: Unit = {
-    val xs3 = mutable.ListBuffer(Right("foo"), Left(0), Right("bar"))
+    val xs3: mutable.ListBuffer[Either[Int, String]] = mutable.ListBuffer(Right("foo"), Left(0), Right("bar"))
```

Comment:
If dotty initially infers `Right[Nothing, String] | Left[Int, Nothing]` and has a rule that: "if the type needs to be inferred, look for supertypes that avoid `Nothing`", we'll get `Either[Int, String]` which is what we want.  If this is too greedy, one could restrict it to only operate on enums.  (It should probably be symmetric with Any in contravariant position.)